src/software/Localization/main_rigLocalizer.cpp

```diff
@@ -240,9 +259,31 @@ int main(int argc, char** argv)
       POPART_COUT("\tnNearestKeyFrames: " << nNearestKeyFrames);
     }
 #endif
-
+    
+  }
+  
+  // check for consistency of the estimators to use
+  // only loransac and acransac are currently supported
+  if(robust::EROBUST_ESTIMATOR_stringToEnum(str_estimatorType) != robust::EROBUST_ESTIMATOR::ROBUST_ESTIMATOR_LORANSAC &&
+     robust::EROBUST_ESTIMATOR_stringToEnum(str_estimatorType) != robust::EROBUST_ESTIMATOR::ROBUST_ESTIMATOR_ACRANSAC)
+  {
+    POPART_CERR("Only " << robust::EROBUST_ESTIMATOR::ROBUST_ESTIMATOR_ACRANSAC 
+            << " and " << robust::EROBUST_ESTIMATOR::ROBUST_ESTIMATOR_LORANSAC 
+            << " are supported!");
+    return EXIT_FAILURE;
+  }
+  // for loransac we need thresholds > 0
+  if(robust::EROBUST_ESTIMATOR_stringToEnum(str_estimatorType) == robust::EROBUST_ESTIMATOR::ROBUST_ESTIMATOR_LORANSAC)
```

Comment:
I don't think, it's a good thing to spread the logic details into many places (makes it difficult to read & maintain).
If it doesn't make sense, it will throw an exception with an explicit message in the loransac function.